docs/modules/ROOT/pages/testing.adoc

```diff
@@ -0,0 +1,624 @@
+
+== Maven Dependency
+
+To use the `ScorerExtension`, include the following Maven dependency in your pom.xml:
+
+[source, xml]
+----
+<dependency>
+    <groupId>io.quarkiverse.langchain4j</groupId>
+    <artifactId>quarkus-langchain4j-testing-scorer-junit5</artifactId>
+    <scope>test</scope>
+</dependency>
+----
+
+== Using the extension
+
+To use the extension, annotate your test class with `@ExtendWith(ScorerExtension.class)` or `@AiScorer`:
+
+[source,java]
+----
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import io.quarkiverse.langchain4j.scorer.junit5.ScorerExtension;
+
+@ExtendWith(ScorerExtension.class)
+public class MyScorerTests {
+
+    // Test cases go here
+}
+----
+
+Or, you can use the `@AiScorer` annotation:
+
+[source,java]
+----
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import io.quarkiverse.langchain4j.scorer.junit5.AiScorer;
+
+@AiScorer
+public class MyScorerTests {
+
+    // Test cases go here
+}
+----
+
+This Junit5 extension can be combined with `@QuarkusTest` to test Quarkus applications:
+
+[source,java]
+----
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import io.quarkiverse.langchain4j.scorer.junit5.AiScorer;
+
+@QuarkusTest
+@AiScorer
+public class MyScorerTests {
+
+    // Test cases go here
+}
+----
+
+== Concepts
+
+=== Scorer
+
+The `Scorer` (`io.quarkiverse.langchain4j.testing.scorer.Scorer`) is a utility that evaluates a _set of samples_ (represented by `io.quarkiverse.langchain4j.testing.scorer.Samples`) against a function (parts of the application) and a set of evaluation strategies.
+It can run evaluations concurrently and produces an `EvaluationReport` summarizing the results and providing the _score_.
+
+The _score_ is the percentage of passed evaluations (between 0.0 and 100.0).
+It is calculated as the ratio of the number of passed evaluations to the total number of evaluations.
+
+In general, tests using the `Scorer` follow this pattern:
+
+[source, java]
+----
+ @Test
+void testAiService(@ScorerConfiguration(concurrency = 5) Scorer scorer, // The scorer instance, with concurrency set to 5
+                   @SampleLocation("src/test/resources/samples.yaml") Samples<String> samples) { // The samples loaded from a YAML file
+    // `aiServiceEvaluation` is a function that takes the sample inputs and returns an output - this is the evaluated output
+    EvaluationReport report = scorer.evaluate(samples, aiServiceEvaluation,
+            new SemanticSimilarityStrategy(0.8)); // The evaluation strategy
+    assertThat(report.score()).isGreaterThanOrEqualTo(70); // Assert the score
+}
+----
+
+=== Samples
+
+A `Sample` (`io.quarkiverse.langchain4j.testing.scorer.EvaluationSample`) represents a single input-output test case.
+It includes:
+- a name: the name of the sample,
+- the inputs: The input data for the test,
```

Comment:
I would drop the capitalization here and for the items below or be consistent for the first item (but I recommend the former).

Reply:
Fixed.